wip-0019.md

```diff
@@ -0,0 +1,182 @@
+protocol follows in the traditional way, with the only addition of a new reducer function that concatenates the
+revealed values in a deterministic order and hashes the sequence together so as to produce a single, uniform and
+pseudo-random output that will be used as the result of the request.
+
+Additionally, a new deterministic ordering for reveals inside tally transactions is introduced so as to rule out any
+malleability attempt from tally producers. 
+
+Due to the overlap of Witnet's inherent commitment scheme with this last aggregation step, there is no need for the
+committed bytes to be verifiably random. Under the assumptions of stock hashing functions, even if the committers
+collude around crafting their commitments in such a way that the output is favorable to them, as long as 1 single
+committer remains independent and commits a value that is secret to the rest, the final result will hold the desired
+randomness properties.
+
+## Specification
+
+### Introduction of data request type `1`
+
+**(1)** A new type of data source is introduced with name `Rng` and identifier `1`.
+
+**(2)** Type `1` sources have the same `script` attribute as type `0` sources, but no `url` attribute.
+
+**(3)** The script field in type `1` sources use the same RADON operators and encoding as type `0`.
+
+### Retrieval of type `1` sources
+
+**(4)** When retrieving a type `1` source, a node MUST generate on the spot a random and secret sequence of 32
+bytes.
+
+**(5)** This random and secret sequence of 32 bytes MUST be used as the result of the retrieval. 
+
+### Reveal of data requests containing type `1` sources
+
+**(6)** Data requests containing type `1` sources MUST be revealed normally.
+
+### Ordering of reveals in tally transactions
+
+**(7)** When producing and validating tally transactions, reveals must be listed and considered in increasing order
+of the big-endian value of the `SHA256` hash of the concatenation of each revealer's 20-bytes Public Key Hash followed
+by the 32-bytes ID of the data request.
+
+```rust
+// Rust-alike pseudocode
+reveals.sort(|reveal| {
+  let mut data = Vec::new();
+  data.extend(reveal.body.pkh.bytes);
+  data.extend(data_request_id.bytes);
+
+  sha256(data)
+});
+```
+
+### Addition of a `ConcatenateAndHash` reducer
+
+**(8)** A new reducer function with code `0x0B` and name `ConcatenateAndHash` is introduced.
+
+**(9)** The `ConcatenateAndHash` reducer function is a general purpose reducer function: it can be used as a reducer
+in the RADON scripts of any type of data sources, as well as in the reducing step of the aggregation and tally stages.
+
+**(10)** The `ConcatenateAndHash` reducer function only accepts a non-empty homogeneous `Array` of byte sequences as its
+input. That is, an `Array` containing one or more items of the `Bytes` type.
+
+**(11)** May the `ConcatenateAndHash` reducer function be applied on an input not satisfying (9), output MUST be an
```

Comment:
I don't understand what does `(9)` refer to. Perhaps it's a typo?

```suggestion
**(11)** May the `ConcatenateAndHash` reducer function be applied on an input not satisfying (10), output MUST be an
```

Reply:
True, I'll fix it. Thanks!

Reply:
Resolved